Get AssertionError when submitting url post to selftext only subreddit while ratelimited

**Describe the bug**
<!-- A clear and concise description of what the bug is. -->
If I am ratelimted on Reddit, such as for posting to /r/tifu, and I post a URL post, which is banned, instead of getting either error, I get an AssertionError

**To Reproduce**
Steps to reproduce the behavior:
Submit to a selftext-only subreddit while post-ratelimited. 

**Expected behavior**
<!-- A clear and concise description of what you expected to happen. -->
Raise one or both errors

**Code/Logs**
<!-- include your code, without the Reddit() initialization, so as to not leak private credentials. -->

Code:
```python
reddit.subreddit("tifu").submit("Test", url="https://www.google.com", validate_on_submit=True)
```

Error:
```python
  File "<ipython-input-8-2dd185cc0bda>", line 1, in <module>
    reddit.subreddit("tifu").submit("Test", url="https://www.google.com", validate_on_submit=True)
  File "~/PycharmProjects/praw/praw/models/reddit/subreddit.py", line 835, in submit
    return self._reddit.post(API_PATH["submit"], data=data)
  File "~/PycharmProjects/praw/praw/reddit.py", line 560, in post
    return self._objector.objectify(data)
  File "~/PycharmProjects/praw/praw/objector.py", line 172, in objectify
    errors,
AssertionError: ('Errors did not get raised as an APIException', [['NO_LINKS', 'that subreddit only allows text posts', 'sr'], ['RATELIMIT', 'you are doing that too much. try again in 13 minutes.', 'ratelimit']])
```

**System Info**
 - OS: MacOS 10.14
 - Python: 3.7
 - PRAW Version: 7.0.0dev0

--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -51,16 +51,20 @@ Unreleased
 
 .. warning:: In May-June 2020, Reddit will force all submissions to run through
   a subreddit's validation rules.
-* Introduced a data class, :class:`.RedditErrorItem`, to represent an individual
-  error item returned from Reddit.
+* Introduced a data class, :class:`.RedditErrorItem`, to represent an
+  individual error item returned from Reddit.
 * Class :class:`.RedditAPIException` now serves as a container for the
-  :class:`.RedditErrorItem`\ s. You can obtain the individual items by either
-  iterating over the :class:`.RedditAPIException` or using index numbers.
-* ``APIException`` is an alias to :class:`.RedditAPIException`.
+  :class:`.RedditErrorItem`\ s. You can access the items by doing
+  ``RedditAPIException.items``, which returns a list.
+* :class:`.APIException` is an alias to :class:`.RedditAPIException`.
+
+**Deprecated**
+
+* :class:`.APIException` is deprecated and slotted for removal in PRAW 8.0.
 
 **Removed**
 
-* Converting :class:`.RedditAPIException` to string will no longer escape unicode
+* Converting :class:`.APIException` to string will no longer escape unicode
   characters.
 * Module ``praw.models.modaction`` no longer exists. Please use the module
   ``praw.models.mod_action``, or directly import ``ModAction``
@@ -138,7 +142,7 @@ Unreleased
 
 **Expected Changes**
 
-* The behavior of :class:`.RedditAPIException` will no longer unicode-escape strings
+* The behavior of func:`APIException` will no longer unicode-escape strings
   in the next minor release
 
 6.4.0 (2019/09/21)
@@ -596,7 +600,7 @@ parameter as described below:
 
 * Uploading an image resulting in too large of a request (>500 KB) now
   raises ``prawcore.TooLarge`` instead of an ``AssertionError``.
-* Uploading an invalid image raises :class:`.RedditAPIException`.
+* Uploading an invalid image raises func:`APIException`.
 * :class:`.Redditor` instances obtained via :attr:`.moderator` (e.g.,
   ``reddit.subreddit('subreddit').moderator()``) will contain attributes with
   the relationship metadata (e.g., ``mod_permissions``).

--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -51,15 +51,16 @@ Unreleased
 
 .. warning:: In May-June 2020, Reddit will force all submissions to run through
   a subreddit's validation rules.
-* Introduced a new class, :class:`.RedditErrorItem`, to represent an individual
+* Introduced a data class, :class:`.RedditErrorItem`, to represent an individual
   error item returned from Reddit.
-* Class :class:`.APIException` now serves as a container for the
-  :class:`.RedditErrorItem` s. You can obtain the individual items by either
-  iterating over the :class:`.APIException` or using index numbers.
+* Class :class:`.RedditAPIException` now serves as a container for the
+  :class:`.RedditErrorItem`\ s. You can obtain the individual items by either
+  iterating over the :class:`.RedditAPIException` or using index numbers.
+* ``APIException`` is an alias to :class:`.RedditAPIException`.
 
 **Removed**
 
-* Converting :class:`.APIException` to string will no longer escape unicode
+* Converting :class:`.RedditAPIException` to string will no longer escape unicode
   characters.
 * Module ``praw.models.modaction`` no longer exists. Please use the module
   ``praw.models.mod_action``, or directly import ``ModAction``
@@ -137,7 +138,7 @@ Unreleased
 
 **Expected Changes**
 
-* The behavior of :class:`.APIException` will no longer unicode-escape strings
+* The behavior of :class:`.RedditAPIException` will no longer unicode-escape strings
   in the next minor release
 
 6.4.0 (2019/09/21)
@@ -595,7 +596,7 @@ parameter as described below:
 
 * Uploading an image resulting in too large of a request (>500 KB) now
   raises ``prawcore.TooLarge`` instead of an ``AssertionError``.
-* Uploading an invalid image raises :class:`.APIException`.
+* Uploading an invalid image raises :class:`.RedditAPIException`.
 * :class:`.Redditor` instances obtained via :attr:`.moderator` (e.g.,
   ``reddit.subreddit('subreddit').moderator()``) will contain attributes with
   the relationship metadata (e.g., ``mod_permissions``).

--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -51,6 +51,7 @@ Unreleased
 
 .. warning:: In May-June 2020, Reddit will force all submissions to run through
   a subreddit's validation rules.
+
 * Introduced a data class, :class:`.RedditErrorItem`, to represent an
   individual error item returned from Reddit.
 * Class :class:`.RedditAPIException` now serves as a container for the

--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -51,6 +51,11 @@ Unreleased
 
 .. warning:: In May-June 2020, Reddit will force all submissions to run through
   a subreddit's validation rules.
+* Introduced a new class, :class:`.RedditErrorItem`, to represent an individual
+  error item returned from Reddit.
+* Class :class:`.APIException` now serves as a container for the
+  :class:`.RedditErrorItem` s. You can obtain the individual items by either
+  iterating over the :class:`.APIException` or using index numbers.
 
 **Removed**
 

--- a/docs/index.rst
+++ b/docs/index.rst
@@ -54,6 +54,7 @@ applications and **web** applications.
 
    package_info/change_log
    package_info/contributing
+   package_info/praw7_migration
    package_info/references
 
 .. toctree::

--- a/docs/package_info/praw7_migration.rst
+++ b/docs/package_info/praw7_migration.rst
@@ -25,7 +25,7 @@ attributes are accessed.
 
     try:
         reddit.subreddit("test").submit("Test Title", url="invalidurl")
-    except RedditAPIException as exception:
+    except APIException as exception:
         print(exception.error_type)
 
 This can generally be changed to

--- a/docs/package_info/praw7_migration.rst
+++ b/docs/package_info/praw7_migration.rst
@@ -4,17 +4,17 @@ Migrating to PRAW 7.X
 Exception Handling
 ------------------
 
-Class ``APIException`` has also been renamed to :class:`.RedditAPIException`.
-Importing ``APIException`` will still work, but is deprecated, but will be
-removed from PRAW 8.0.
+Class :class:`.APIException` has also been renamed to
+:class:`.RedditAPIException`. Importing :class:`.APIException` will still work,
+but is deprecated, but will be removed in PRAW 8.0.
 
 PRAW 7 introduced a fundamental change in how exceptions are recieved from
-Reddit's API.Reddit can return multiple exceptions for one API action, and
+Reddit's API. Reddit can return multiple exceptions for one API action, and
 as such, the exception :class:`.RedditAPIException` serves as a container for
 each of the true exception objects. These objects are instances of
 :class:`.RedditErrorItem`, and they contain the information of one "error" from
-Reddit's API. They have the three data attributes that
-:class:`.RedditAPIException` used to contain.
+Reddit's API. They have the three data attributes that :class:`.APIException`
+used to contain.
 
 Most code regarding exceptions can be quickly fixed to work under the new
 system. All of the exceptions are stored in the ``items`` attribute of the

--- a/docs/package_info/praw7_migration.rst
+++ b/docs/package_info/praw7_migration.rst
@@ -4,22 +4,28 @@ Migrating to PRAW 7.X
 Exception Handling
 ------------------
 
+Class ``APIException`` has also been renamed to :class:`.RedditAPIException`.
+Importing ``APIException`` will still work, but is deprecated, but will be
+removed from PRAW 8.0.
+
 PRAW 7 introduced a fundamental change in how exceptions are recieved from
-Reddit's API. Reddit can return multiple exceptions for one API action, and
-as such, the previous :class:`.APIException` serves as a container for each of
-the true exception objects. These objects are instances of
+Reddit's API.Reddit can return multiple exceptions for one API action, and
+as such, the exception :class:`.RedditAPIException` serves as a container for
+each of the true exception objects. These objects are instances of
 :class:`.RedditErrorItem`, and they contain the information of one "error" from
-Reddit's API. They have the three data attributes that :class:`.APIException`
-used to contain.
+Reddit's API. They have the three data attributes that
+:class:`.RedditAPIException` used to contain.
 
 Most code regarding exceptions can be quickly fixed to work under the new
-system. In the example code below, observe how attributes are accessed.
+system. All of the exceptions are stored in the ``items`` attribute of the
+exception as entries in a list. In the example code below, observe how
+attributes are accessed.
 
 .. code-block:: python
 
     try:
         reddit.subreddit("test").submit("Test Title", url="invalidurl")
-    except APIException as exception:
+    except RedditAPIException as exception:
         print(exception.error_type)
 
 This can generally be changed to
@@ -28,8 +34,8 @@ This can generally be changed to
 
     try:
         reddit.subreddit("test").submit("Test Title", url="invalidurl")
-    except APIException as exception:
-        print(exception[0].error_type)
+    except RedditAPIException as exception:
+        print(exception.items[0].error_type)
 
 However, this should not be done, as this will only work for one error. The
 probability of Reddit's API returning multiple exceptions, especially on
@@ -40,8 +46,8 @@ and do the action on each item in the iterator.
 
     try:
         reddit.subreddit("test").submit("Test Title", url="invalidurl")
-    except APIException as exception:
-        for subexception in exception:
+    except RedditAPIException as exception:
+        for subexception in exception.items:
             print(subexception.error_type)
 
 Alternatively, the exceptions are provided to the exception constructor, so
@@ -52,5 +58,5 @@ exceptions.
 
     try:
         reddit.subreddit("test").submit("Test Title", url="invalidurl")
-    except APIException as exception:
+    except RedditAPIException as exception:
         print(exception)

--- a/docs/package_info/praw7_migration.rst
+++ b/docs/package_info/praw7_migration.rst
@@ -1,0 +1,56 @@
+Migrating to PRAW 7.X
+=====================
+
+Exception Handling
+------------------
+
+PRAW 7 introduced a fundamental change in how exceptions are recieved from
+Reddit's API. Reddit can return multiple exceptions for one API action, and
+as such, the previous :class:`.APIException` serves as a container for each of
+the true exception objects. These objects are instances of
+:class:`.RedditErrorItem`, and they contain the information of one "error" from
+Reddit's API. They have the three data attributes that :class:`.APIException`
+used to contain.
+
+Most code regarding exceptions can be quickly fixed to work under the new
+system. In the example code below, observe how attributes are accessed.
+
+.. code-block:: python
+
+    try:
+        reddit.subreddit("test").submit("Test Title", url="invalidurl")
+    except APIException as exception:
+        print(exception.error_type)
+
+This can generally be changed to
+
+.. code-block:: python
+
+    try:
+        reddit.subreddit("test").submit("Test Title", url="invalidurl")
+    except APIException as exception:
+        print(exception[0].error_type)
+
+However, this should not be done, as this will only work for one error. The
+probability of Reddit's API returning multiple exceptions, especially on
+submit actions, should be addressed. Rather, iterate over the exception,
+and do the action on each item in the iterator.
+
+.. code-block:: python
+
+    try:
+        reddit.subreddit("test").submit("Test Title", url="invalidurl")
+    except APIException as exception:
+        for subexception in exception:
+            print(subexception.error_type)
+
+Alternatively, the exceptions are provided to the exception constructor, so
+printing the exception directly will also allow you to see all of the
+exceptions.
+
+.. code-block:: python
+
+    try:
+        reddit.subreddit("test").submit("Test Title", url="invalidurl")
+    except APIException as exception:
+        print(exception)

--- a/praw/exceptions.py
+++ b/praw/exceptions.py
@@ -7,31 +7,84 @@ wrong on the client side. Both of these classes extend :class:`.PRAWException`.
 All other exceptions are subclassed from :class:`.ClientException`.
 
 """
-from typing import Optional
+from typing import List, Optional, TypeVar, Union
+
+_RedditErrorItem = TypeVar("_RedditErrorItem")
 
 
 class PRAWException(Exception):
     """The base PRAW Exception that all other exception classes extend."""
 
 
-class APIException(PRAWException):
-    """Indicate exception that involve responses from Reddit's API."""
+class RedditErrorItem:
+    """Represents a single error returned from Reddit's API."""
 
-    def __init__(self, error_type: str, message: str, field: Optional[str]):
-        """Initialize an instance of APIException.
+    @property
+    def error_message(self):
+        """Get the completed error message string."""
+        error_str = "{}: '{}'".format(self.error_type, self.message)
+        if self.field:
+            error_str += " on field '{}'".format(self.field)
+        return error_str
+
+    def __init__(
+        self, error_type: str, message: str, field: Optional[str] = None
+    ):
+        """Instantize an error item.
 
         :param error_type: The error type set on Reddit's end.
         :param message: The associated message for the error.
         :param field: The input field associated with the error if available.
         """
-        error_str = "{}: '{}'".format(error_type, message)
-        if field:
-            error_str += " on field '{}'".format(field)
-
-        super().__init__(error_str)
         self.error_type = error_type
         self.message = message
         self.field = field
+
+    def __eq__(self, other: Union[_RedditErrorItem, List[str]]):
+        """Check for equality."""
+        if isinstance(other, RedditErrorItem):
+            return (self.error_type, self.message, self.field) == (
+                other.error_type,
+                other.message,
+                other.field,
+            )
+        elif isinstance(other, list):
+            return [self.error_type, self.message, self.field] == other
+        return False
+
+    def __repr__(self):
+        """Get the message returned from repr(self)."""
+        return self.error_message
+
+    def __str__(self):
+        """Get the message returned from str(self)."""
+        return self.error_message
+
+
+class APIException(PRAWException):
+    """Container for error messages from Reddit's API."""
+
+    def __init__(self, items: List[Union[RedditErrorItem, List[str]]]):
+        """Initialize an instance of APIException.
+
+        :param items: Either a list of instances of :class:`.RedditErrorItem`
+            or a list containing lists of unformed errors.
+        """
+        self.items = []
+        for item in items:
+            if isinstance(item, RedditErrorItem):
+                self.items.append(item)
+            else:
+                self.items.append(RedditErrorItem(*item))
+        super().__init__(*self.items)
+
+    def __iter__(self):
+        """Iterate over the list of items."""
+        return iter(self.items)
+
+    def __getitem__(self, item: Union[int, slice]):
+        """Get an item or slice of items from the list of items."""
+        return self.items[item]
 
 
 class ClientException(PRAWException):

--- a/praw/exceptions.py
+++ b/praw/exceptions.py
@@ -9,6 +9,7 @@ All other exceptions are subclassed from :class:`.ClientException`.
 
 """
 from typing import List, Optional, TypeVar, Union
+from warnings import warn
 
 _RedditErrorItem = TypeVar("_RedditErrorItem")
 
@@ -23,9 +24,9 @@ class RedditErrorItem:
     @property
     def error_message(self):
         """Get the completed error message string."""
-        error_str = "{}: '{}'".format(self.error_type, self.message)
+        error_str = "{}: {!r}".format(self.error_type, self.message)
         if self.field:
-            error_str += " on field '{}'".format(self.field)
+            error_str += " on field {!r}".format(self.field)
         return error_str
 
     def __init__(
@@ -49,7 +50,7 @@ class RedditErrorItem:
                 other.message,
                 other.field,
             )
-        return NotImplemented
+        return super().__eq__(other)
 
     def __repr__(self):
         """Return repr(self)."""
@@ -66,7 +67,9 @@ class RedditAPIException(PRAWException):
     """Container for error messages from Reddit's API."""
 
     @staticmethod
-    def parse_exception_list(exclist: List[Union[RedditErrorItem, List[str]]]):
+    def parse_exception_list(
+        exceptions: List[Union[RedditErrorItem, List[str]]]
+    ):
         """Covert an exception list into a :class:`.RedditErrorItem` list."""
         return [
             exception
@@ -76,7 +79,7 @@ class RedditAPIException(PRAWException):
                 exception[1],
                 exception[2] if bool(exception[2]) else "",
             )
-            for exception in exclist
+            for exception in exceptions
         ]
 
     def __init__(self, items: List[Union[RedditErrorItem, List[str]]]):
@@ -89,7 +92,45 @@ class RedditAPIException(PRAWException):
         super().__init__(*self.items)
 
 
-APIException = RedditAPIException
+class APIException(RedditAPIException):
+    """Old class preserved for alias purposes."""
+
+    @property
+    def error_type(self) -> str:
+        """Get error_type."""
+        return self._get_old_attr("error_type")
+
+    @property
+    def message(self) -> str:
+        """Get message."""
+        return self._get_old_attr("message")
+
+    @property
+    def field(self) -> str:
+        """Get field."""
+        return self._get_old_attr("field")
+
+    def _get_old_attr(self, attrname):
+        warn(
+            "Accessing attribute ``{}`` through APIException is deprecated. "
+            "This behavior will be removed in PRAW 8.0. Check out "
+            "https://praw.readthedocs.io/en/latest/package_info/"
+            "praw7_migration.html to learn how to migrate your code.".format(
+                attrname
+            ),
+            category=DeprecationWarning,
+            stacklevel=3,
+        )
+        return getattr(self.items[0], attrname)
+
+    def __init__(self, arg1, *optional_args):
+        """Initialize the exception."""
+        if isinstance(arg1, str):
+            super().__init__([[arg1, *optional_args]])
+        elif isinstance(arg1, list) and isinstance(arg1[0], str):
+            super().__init__([arg1])
+        else:
+            super().__init__(arg1)
 
 
 class ClientException(PRAWException):

--- a/praw/exceptions.py
+++ b/praw/exceptions.py
@@ -63,8 +63,8 @@ class RedditErrorItem:
         return self.error_message
 
 
-class RedditAPIException(PRAWException):
-    """Container for error messages from Reddit's API."""
+class APIException(PRAWException):
+    """Old class preserved for alias purposes."""
 
     @staticmethod
     def parse_exception_list(
@@ -130,8 +130,8 @@ class RedditAPIException(PRAWException):
         super().__init__(*self.items)
 
 
-class APIException(RedditAPIException):
-    """Old class preserved for alias purposes."""
+class RedditAPIException(APIException):
+    """Container for error messages from Reddit's API."""
 
 
 class ClientException(PRAWException):

--- a/praw/exceptions.py
+++ b/praw/exceptions.py
@@ -1,8 +1,9 @@
 """PRAW exception classes.
 
-Includes two main exceptions: :class:`.APIException` for when something goes
-wrong on the server side, and :class:`.ClientException` when something goes
-wrong on the client side. Both of these classes extend :class:`.PRAWException`.
+Includes two main exceptions: :class:`.RedditAPIException` for when something
+goes wrong on the server side, and :class:`.ClientException` when something
+goes wrong on the client side. Both of these classes extend
+:class:`.PRAWException`.
 
 All other exceptions are subclassed from :class:`.ClientException`.
 
@@ -30,11 +31,11 @@ class RedditErrorItem:
     def __init__(
         self, error_type: str, message: str, field: Optional[str] = None
     ):
-        """Instantize an error item.
+        """Instantiate an error item.
 
         :param error_type: The error type set on Reddit's end.
         :param message: The associated message for the error.
-        :param field: The input field associated with the error if available.
+        :param field: The input field associated with the error, if available.
         """
         self.error_type = error_type
         self.message = message
@@ -48,43 +49,47 @@ class RedditErrorItem:
                 other.message,
                 other.field,
             )
-        elif isinstance(other, list):
-            return [self.error_type, self.message, self.field] == other
-        return False
+        return NotImplemented
 
     def __repr__(self):
-        """Get the message returned from repr(self)."""
-        return self.error_message
+        """Return repr(self)."""
+        return "{}(error_type={!r}, message={!r}, field={!r})".format(
+            self.__class__.__name__, self.error_type, self.message, self.field
+        )
 
     def __str__(self):
         """Get the message returned from str(self)."""
         return self.error_message
 
 
-class APIException(PRAWException):
+class RedditAPIException(PRAWException):
     """Container for error messages from Reddit's API."""
 
+    @staticmethod
+    def parse_exception_list(exclist: List[Union[RedditErrorItem, List[str]]]):
+        """Covert an exception list into a :class:`.RedditErrorItem` list."""
+        return [
+            exception
+            if isinstance(exception, RedditErrorItem)
+            else RedditErrorItem(
+                exception[0],
+                exception[1],
+                exception[2] if bool(exception[2]) else "",
+            )
+            for exception in exclist
+        ]
+
     def __init__(self, items: List[Union[RedditErrorItem, List[str]]]):
-        """Initialize an instance of APIException.
+        """Initialize an instance of RedditAPIException.
 
         :param items: Either a list of instances of :class:`.RedditErrorItem`
             or a list containing lists of unformed errors.
         """
-        self.items = []
-        for item in items:
-            if isinstance(item, RedditErrorItem):
-                self.items.append(item)
-            else:
-                self.items.append(RedditErrorItem(*item))
+        self.items = self.parse_exception_list(items)
         super().__init__(*self.items)
 
-    def __iter__(self):
-        """Iterate over the list of items."""
-        return iter(self.items)
 
-    def __getitem__(self, item: Union[int, slice]):
-        """Get an item or slice of items from the list of items."""
-        return self.items[item]
+APIException = RedditAPIException
 
 
 class ClientException(PRAWException):

--- a/praw/exceptions.py
+++ b/praw/exceptions.py
@@ -82,19 +82,6 @@ class RedditAPIException(PRAWException):
             for exception in exceptions
         ]
 
-    def __init__(self, items: List[Union[RedditErrorItem, List[str]]]):
-        """Initialize an instance of RedditAPIException.
-
-        :param items: Either a list of instances of :class:`.RedditErrorItem`
-            or a list containing lists of unformed errors.
-        """
-        self.items = self.parse_exception_list(items)
-        super().__init__(*self.items)
-
-
-class APIException(RedditAPIException):
-    """Old class preserved for alias purposes."""
-
     @property
     def error_type(self) -> str:
         """Get error_type."""
@@ -123,14 +110,28 @@ class APIException(RedditAPIException):
         )
         return getattr(self.items[0], attrname)
 
-    def __init__(self, arg1, *optional_args):
-        """Initialize the exception."""
-        if isinstance(arg1, str):
-            super().__init__([[arg1, *optional_args]])
-        elif isinstance(arg1, list) and isinstance(arg1[0], str):
-            super().__init__([arg1])
-        else:
-            super().__init__(arg1)
+    def __init__(
+        self,
+        items: Union[List[Union[RedditErrorItem, List[str], str]], str],
+        *optional_args: str
+    ):
+        """Initialize an instance of RedditAPIException.
+
+        :param items: Either a list of instances of :class:`.RedditErrorItem`
+            or a list containing lists of unformed errors.
+        :param optional_args: Takes the second and third arguments that
+            :class:`.APIException` used to take.
+        """
+        if isinstance(items, str):
+            items = [[items, *optional_args]]
+        elif isinstance(items, list) and isinstance(items[0], str):
+            items = [items]
+        self.items = self.parse_exception_list(items)
+        super().__init__(*self.items)
+
+
+class APIException(RedditAPIException):
+    """Old class preserved for alias purposes."""
 
 
 class ClientException(PRAWException):

--- a/praw/models/reddit/live.py
+++ b/praw/models/reddit/live.py
@@ -77,7 +77,7 @@ class LiveContributorRelationship:
     ):
         """Invite a redditor to be a contributor of the live thread.
 
-        Raise :class:`praw.exceptions.APIException` if the invitation
+        Raise :class:`praw.exceptions.RedditAPIException` if the invitation
         already exists.
 
         :param redditor: A redditor name (e.g., ``'spez'``) or

--- a/praw/models/reddit/mixins/reportable.py
+++ b/praw/models/reddit/mixins/reportable.py
@@ -10,7 +10,7 @@ class ReportableMixin:
 
         :param reason: The reason for reporting.
 
-        Raises :class:`.APIException` if ``reason`` is longer than 100
+        Raises :class:`.RedditAPIException` if ``reason`` is longer than 100
         characters.
 
         Example usage:

--- a/praw/models/reddit/subreddit.py
+++ b/praw/models/reddit/subreddit.py
@@ -2991,7 +2991,7 @@ class SubredditStylesheet:
                     "BAD_CSS_NAME",
                     "IMAGE_ERROR",
                 ], "Please file a bug with PRAW"
-                raise APIException(error_type, error_value, None)
+                raise APIException([[error_type, error_value, None]])
             return response
 
     def _upload_style_asset(self, image_path, image_type):

--- a/praw/models/reddit/subreddit.py
+++ b/praw/models/reddit/subreddit.py
@@ -13,9 +13,9 @@ from prawcore import Redirect
 
 from ...const import API_PATH, JPEG_HEADER
 from ...exceptions import (
-    RedditAPIException,
     ClientException,
     InvalidFlairTemplateID,
+    RedditAPIException,
     TooLargeMediaException,
     WebSocketException,
 )

--- a/praw/models/reddit/subreddit.py
+++ b/praw/models/reddit/subreddit.py
@@ -13,7 +13,7 @@ from prawcore import Redirect
 
 from ...const import API_PATH, JPEG_HEADER
 from ...exceptions import (
-    APIException,
+    RedditAPIException,
     ClientException,
     InvalidFlairTemplateID,
     TooLargeMediaException,
@@ -2991,7 +2991,7 @@ class SubredditStylesheet:
                     "BAD_CSS_NAME",
                     "IMAGE_ERROR",
                 ], "Please file a bug with PRAW"
-                raise APIException([[error_type, error_value, None]])
+                raise RedditAPIException([[error_type, error_value, None]])
             return response
 
     def _upload_style_asset(self, image_path, image_type):
@@ -3158,7 +3158,7 @@ class SubredditStylesheet:
 
         Raises ``prawcore.TooLarge`` if the overall request body is too large.
 
-        Raises :class:`.APIException` if there are other issues with the
+        Raises :class:`.RedditAPIException` if there are other issues with the
         uploaded image. Unfortunately the exception info might not be very
         specific, so try through the website with the same image to see what
         the problem actually might be.
@@ -3181,7 +3181,7 @@ class SubredditStylesheet:
 
         Raises ``prawcore.TooLarge`` if the overall request body is too large.
 
-        Raises :class:`.APIException` if there are other issues with the
+        Raises :class:`.RedditAPIException` if there are other issues with the
         uploaded image. Unfortunately the exception info might not be very
         specific, so try through the website with the same image to see what
         the problem actually might be.
@@ -3206,7 +3206,7 @@ class SubredditStylesheet:
 
         Raises ``prawcore.TooLarge`` if the overall request body is too large.
 
-        Raises :class:`.APIException` if there are other issues with the
+        Raises :class:`.RedditAPIException` if there are other issues with the
         uploaded image. Unfortunately the exception info might not be very
         specific, so try through the website with the same image to see what
         the problem actually might be.
@@ -3243,7 +3243,7 @@ class SubredditStylesheet:
 
         Raises ``prawcore.TooLarge`` if the overall request body is too large.
 
-        Raises :class:`.APIException` if there are other issues with the
+        Raises :class:`.RedditAPIException` if there are other issues with the
         uploaded image. Unfortunately the exception info might not be very
         specific, so try through the website with the same image to see what
         the problem actually might be.
@@ -3268,7 +3268,7 @@ class SubredditStylesheet:
 
         Raises ``prawcore.TooLarge`` if the overall request body is too large.
 
-        Raises :class:`.APIException` if there are other issues with the
+        Raises :class:`.RedditAPIException` if there are other issues with the
         uploaded image. Unfortunately the exception info might not be very
         specific, so try through the website with the same image to see what
         the problem actually might be.
@@ -3291,7 +3291,7 @@ class SubredditStylesheet:
 
         Raises ``prawcore.TooLarge`` if the overall request body is too large.
 
-        Raises :class:`.APIException` if there are other issues with the
+        Raises :class:`.RedditAPIException` if there are other issues with the
         uploaded image. Unfortunately the exception info might not be very
         specific, so try through the website with the same image to see what
         the problem actually might be.
@@ -3315,7 +3315,7 @@ class SubredditStylesheet:
 
         Raises ``prawcore.TooLarge`` if the overall request body is too large.
 
-        Raises :class:`.APIException` if there are other issues with the
+        Raises :class:`.RedditAPIException` if there are other issues with the
         uploaded image. Unfortunately the exception info might not be very
         specific, so try through the website with the same image to see what
         the problem actually might be.

--- a/praw/objector.py
+++ b/praw/objector.py
@@ -2,7 +2,7 @@
 
 from typing import Any, Dict, List, Optional, TypeVar, Union
 
-from .exceptions import APIException, ClientException
+from .exceptions import RedditAPIException, ClientException
 from .models.reddit.base import RedditBase
 from .util import snake_case_keys
 
@@ -15,11 +15,11 @@ class Objector:
     @classmethod
     def parse_error(
         cls, data: Union[List[Any], Dict[str, Dict[str, str]]]
-    ) -> Optional[APIException]:
+    ) -> Optional[RedditAPIException]:
         """Convert JSON response into an error object.
 
         :param data: The dict to be converted.
-        :returns: An instance of :class:`~.APIException`, or ``None`` if
+        :returns: An instance of :class:`~.RedditAPIException`, or ``None`` if
             ``data`` doesn't fit this model.
 
         """
@@ -40,7 +40,7 @@ class Objector:
             data,
         )
 
-        return APIException(errors)
+        return RedditAPIException(errors)
 
     @classmethod
     def check_error(cls, data: Union[List[Any], Dict[str, Dict[str, str]]]):
@@ -166,9 +166,9 @@ class Objector:
         if "json" in data and "errors" in data["json"]:
             errors = data["json"]["errors"]
             if len(errors) >= 1:
-                raise APIException(errors)
+                raise RedditAPIException(errors)
             assert not errors, (
-                "Errors did not get raised as an APIException",
+                "Errors did not get raised as an RedditAPIException",
                 errors,
             )
         elif isinstance(data, dict):

--- a/praw/objector.py
+++ b/praw/objector.py
@@ -2,7 +2,7 @@
 
 from typing import Any, Dict, List, Optional, TypeVar, Union
 
-from .exceptions import RedditAPIException, ClientException
+from .exceptions import ClientException, RedditAPIException
 from .models.reddit.base import RedditBase
 from .util import snake_case_keys
 
@@ -35,11 +35,6 @@ class Objector:
         if len(errors) < 1:
             # See `Collection._fetch()`.
             raise ClientException("successful error response", data)
-        assert not len(errors) > 1, (  # Yet to be observed.
-            "multiple error descriptions in response",
-            data,
-        )
-
         return RedditAPIException(errors)
 
     @classmethod
@@ -165,12 +160,8 @@ class Objector:
             return parser.parse(data["json"]["data"], self._reddit)
         if "json" in data and "errors" in data["json"]:
             errors = data["json"]["errors"]
-            if len(errors) >= 1:
+            if len(errors) > 0:
                 raise RedditAPIException(errors)
-            assert not errors, (
-                "Errors did not get raised as an RedditAPIException",
-                errors,
-            )
         elif isinstance(data, dict):
             return self._objectify_dict(data)
 

--- a/praw/objector.py
+++ b/praw/objector.py
@@ -40,7 +40,7 @@ class Objector:
             data,
         )
 
-        return APIException(*errors[0])
+        return APIException(errors)
 
     @classmethod
     def check_error(cls, data: Union[List[Any], Dict[str, Dict[str, str]]]):
@@ -165,8 +165,8 @@ class Objector:
             return parser.parse(data["json"]["data"], self._reddit)
         if "json" in data and "errors" in data["json"]:
             errors = data["json"]["errors"]
-            if len(errors) == 1:
-                raise APIException(*errors[0])
+            if len(errors) >= 1:
+                raise APIException(errors)
             assert not errors, (
                 "Errors did not get raised as an APIException",
                 errors,

--- a/pytest.ini
+++ b/pytest.ini
@@ -1,2 +1,4 @@
 [pytest]
 testpaths = tests
+filterwarnings =
+    ignore::DeprecationWarning

--- a/tests/integration/models/reddit/test_live.py
+++ b/tests/integration/models/reddit/test_live.py
@@ -101,7 +101,7 @@ class TestLiveContributorRelationship(IntegrationTest):
             thread.contributor.invite("nmtake")
             with pytest.raises(APIException) as excinfo:
                 thread.contributor.invite("nmtake")
-        assert excinfo.value.error_type == "LIVEUPDATE_ALREADY_CONTRIBUTOR"
+        assert excinfo.value[0].error_type == "LIVEUPDATE_ALREADY_CONTRIBUTOR"
 
     def test_invite__empty_list(self):
         self.reddit.read_only = False

--- a/tests/integration/models/reddit/test_live.py
+++ b/tests/integration/models/reddit/test_live.py
@@ -3,7 +3,7 @@ import mock
 import pytest
 
 from praw.const import API_PATH
-from praw.exceptions import APIException
+from praw.exceptions import RedditAPIException
 from praw.models import (
     LiveThread,
     LiveUpdate,
@@ -99,9 +99,12 @@ class TestLiveContributorRelationship(IntegrationTest):
             "TestLiveContributorRelationship_test_invite__already_invited"
         ):
             thread.contributor.invite("nmtake")
-            with pytest.raises(APIException) as excinfo:
+            with pytest.raises(RedditAPIException) as excinfo:
                 thread.contributor.invite("nmtake")
-        assert excinfo.value[0].error_type == "LIVEUPDATE_ALREADY_CONTRIBUTOR"
+        assert (
+            excinfo.value.items[0].error_type
+            == "LIVEUPDATE_ALREADY_CONTRIBUTOR"
+        )
 
     def test_invite__empty_list(self):
         self.reddit.read_only = False

--- a/tests/integration/models/reddit/test_subreddit.py
+++ b/tests/integration/models/reddit/test_subreddit.py
@@ -18,7 +18,7 @@ from prawcore import (
 
 from praw.const import PNG_HEADER
 from praw.exceptions import (
-    APIException,
+    RedditAPIException,
     ClientException,
     InvalidFlairTemplateID,
     TooLargeMediaException,
@@ -118,7 +118,7 @@ class TestSubreddit(IntegrationTest):
     def test_create__exists(self):
         self.reddit.read_only = False
         with self.recorder.use_cassette("TestSubreddit.test_create__exists"):
-            with pytest.raises(APIException) as excinfo:
+            with pytest.raises(RedditAPIException) as excinfo:
                 self.reddit.subreddit.create(
                     "redditdev",
                     title="redditdev",
@@ -126,14 +126,14 @@ class TestSubreddit(IntegrationTest):
                     subreddit_type="public",
                     wikimode="disabled",
                 )
-            assert excinfo.value[0].error_type == "SUBREDDIT_EXISTS"
+            assert excinfo.value.items[0].error_type == "SUBREDDIT_EXISTS"
 
     def test_create__invalid_parameter(self):
         self.reddit.read_only = False
         with self.recorder.use_cassette(
             "TestSubreddit.test_create__invalid_parameter"
         ):
-            with pytest.raises(APIException) as excinfo:
+            with pytest.raises(RedditAPIException) as excinfo:
                 # Supplying invalid setting for link_type
                 self.reddit.subreddit.create(
                     name="PRAW_iavynavffv",
@@ -142,14 +142,14 @@ class TestSubreddit(IntegrationTest):
                     subreddit_type="public",
                     wikimode="disabled",
                 )
-            assert excinfo.value[0].error_type == "INVALID_OPTION"
+            assert excinfo.value.items[0].error_type == "INVALID_OPTION"
 
     def test_create__missing_parameter(self):
         self.reddit.read_only = False
         with self.recorder.use_cassette(
             "TestSubreddit.test_create__missing_parameter"
         ):
-            with pytest.raises(APIException) as excinfo:
+            with pytest.raises(RedditAPIException) as excinfo:
                 # Not supplying required field title.
                 self.reddit.subreddit.create(
                     name="PRAW_iavynavffv",
@@ -158,7 +158,7 @@ class TestSubreddit(IntegrationTest):
                     subreddit_type="public",
                     wikimode="disabled",
                 )
-            assert excinfo.value[0].error_type == "NO_TEXT"
+            assert excinfo.value.items[0].error_type == "NO_TEXT"
 
     @mock.patch("time.sleep", return_value=None)
     def test_message(self, _):
@@ -1359,9 +1359,9 @@ class TestSubredditModeration(IntegrationTest):
         with self.recorder.use_cassette(
             "TestSubredditModeration.test_accept_invite__no_invite"
         ):
-            with pytest.raises(APIException) as excinfo:
+            with pytest.raises(RedditAPIException) as excinfo:
                 self.subreddit.mod.accept_invite()
-            assert excinfo.value[0].error_type == "NO_INVITE_FOUND"
+            assert excinfo.value.items[0].error_type == "NO_INVITE_FOUND"
 
     def test_edited(self):
         self.reddit.read_only = False
@@ -1787,11 +1787,11 @@ class TestSubredditRelationships(IntegrationTest):
         with self.recorder.use_cassette(
             "TestSubredditRelationships.moderator_invite__invalid_perm"
         ):
-            with pytest.raises(APIException) as excinfo:
+            with pytest.raises(RedditAPIException) as excinfo:
                 self.subreddit.moderator.invite(
                     self.REDDITOR, permissions=["a"]
                 )
-            assert excinfo.value[0].error_type == "INVALID_PERMISSIONS"
+            assert excinfo.value.items[0].error_type == "INVALID_PERMISSIONS"
 
     @mock.patch("time.sleep", return_value=None)
     def test_moderator_invite__no_perms(self, _):
@@ -2127,22 +2127,22 @@ class TestSubredditStylesheet(IntegrationTest):
         with self.recorder.use_cassette(
             "TestSubredditStylesheet.test_upload__invalid"
         ):
-            with pytest.raises(APIException) as excinfo:
+            with pytest.raises(RedditAPIException) as excinfo:
                 self.subreddit.stylesheet.upload(
                     "praw", self.image_path("invalid.jpg")
                 )
-        assert excinfo.value[0].error_type == "IMAGE_ERROR"
+        assert excinfo.value.items[0].error_type == "IMAGE_ERROR"
 
     def test_upload__invalid_ext(self):
         self.reddit.read_only = False
         with self.recorder.use_cassette(
             "TestSubredditStylesheet.test_upload__invalid_ext"
         ):
-            with pytest.raises(APIException) as excinfo:
+            with pytest.raises(RedditAPIException) as excinfo:
                 self.subreddit.stylesheet.upload(
                     "praw.png", self.image_path("white-square.png")
                 )
-        assert excinfo.value[0].error_type == "BAD_CSS_NAME"
+        assert excinfo.value.items[0].error_type == "BAD_CSS_NAME"
 
     def test_upload__too_large(self):
         self.reddit.read_only = False
@@ -2283,11 +2283,11 @@ class TestSubredditStylesheet(IntegrationTest):
                 "upload_mobile_header",
                 "upload_mobile_icon",
             ]:
-                with pytest.raises(APIException) as excinfo:
+                with pytest.raises(RedditAPIException) as excinfo:
                     getattr(self.subreddit.stylesheet, method)(
                         self.image_path("invalid.jpg")
                     )
-                assert excinfo.value[0].error_type == "IMAGE_ERROR"
+                assert excinfo.value.items[0].error_type == "IMAGE_ERROR"
 
     def test_upload__others_too_large(self):
         self.reddit.read_only = False

--- a/tests/integration/models/reddit/test_subreddit.py
+++ b/tests/integration/models/reddit/test_subreddit.py
@@ -18,10 +18,10 @@ from prawcore import (
 
 from praw.const import PNG_HEADER
 from praw.exceptions import (
-    RedditAPIException,
     ClientException,
     InvalidFlairTemplateID,
     TooLargeMediaException,
+    RedditAPIException,
     WebSocketException,
 )
 from praw.models import (

--- a/tests/integration/models/reddit/test_subreddit.py
+++ b/tests/integration/models/reddit/test_subreddit.py
@@ -20,8 +20,8 @@ from praw.const import PNG_HEADER
 from praw.exceptions import (
     ClientException,
     InvalidFlairTemplateID,
-    TooLargeMediaException,
     RedditAPIException,
+    TooLargeMediaException,
     WebSocketException,
 )
 from praw.models import (

--- a/tests/integration/models/reddit/test_subreddit.py
+++ b/tests/integration/models/reddit/test_subreddit.py
@@ -341,7 +341,7 @@ class TestSubreddit(IntegrationTest):
                 pytest.placeholders.test_subreddit
             )
             with pytest.raises(
-                (APIException, BadRequest)
+                (RedditAPIException, BadRequest)
             ):  # waiting for prawcore fix
                 subreddit.submit(
                     "dfgfdgfdgdf", url="https://www.google.com",
@@ -473,7 +473,7 @@ class TestSubreddit(IntegrationTest):
             )
             image = self.image_path("test.jpg")
             with pytest.raises(
-                (APIException, BadRequest)
+                (RedditAPIException, BadRequest)
             ):  # waiting for prawcore fix
                 subreddit.submit_image(
                     "gdfgfdgdgdgfgfdgdfgfdgfdg", image,
@@ -672,7 +672,7 @@ class TestSubreddit(IntegrationTest):
             )
             image = self.image_path("test.mov")
             with pytest.raises(
-                (APIException, BadRequest)
+                (RedditAPIException, BadRequest)
             ):  # waiting for prawcore fix
                 subreddit.submit_video("gdfgfdgdgdgfgfdgdfgfdgfdg", image)
 

--- a/tests/integration/models/reddit/test_subreddit.py
+++ b/tests/integration/models/reddit/test_subreddit.py
@@ -126,7 +126,7 @@ class TestSubreddit(IntegrationTest):
                     subreddit_type="public",
                     wikimode="disabled",
                 )
-            assert excinfo.value.error_type == "SUBREDDIT_EXISTS"
+            assert excinfo.value[0].error_type == "SUBREDDIT_EXISTS"
 
     def test_create__invalid_parameter(self):
         self.reddit.read_only = False
@@ -142,7 +142,7 @@ class TestSubreddit(IntegrationTest):
                     subreddit_type="public",
                     wikimode="disabled",
                 )
-            assert excinfo.value.error_type == "INVALID_OPTION"
+            assert excinfo.value[0].error_type == "INVALID_OPTION"
 
     def test_create__missing_parameter(self):
         self.reddit.read_only = False
@@ -158,7 +158,7 @@ class TestSubreddit(IntegrationTest):
                     subreddit_type="public",
                     wikimode="disabled",
                 )
-            assert excinfo.value.error_type == "NO_TEXT"
+            assert excinfo.value[0].error_type == "NO_TEXT"
 
     @mock.patch("time.sleep", return_value=None)
     def test_message(self, _):
@@ -1361,7 +1361,7 @@ class TestSubredditModeration(IntegrationTest):
         ):
             with pytest.raises(APIException) as excinfo:
                 self.subreddit.mod.accept_invite()
-            assert excinfo.value.error_type == "NO_INVITE_FOUND"
+            assert excinfo.value[0].error_type == "NO_INVITE_FOUND"
 
     def test_edited(self):
         self.reddit.read_only = False
@@ -1791,7 +1791,7 @@ class TestSubredditRelationships(IntegrationTest):
                 self.subreddit.moderator.invite(
                     self.REDDITOR, permissions=["a"]
                 )
-            assert excinfo.value.error_type == "INVALID_PERMISSIONS"
+            assert excinfo.value[0].error_type == "INVALID_PERMISSIONS"
 
     @mock.patch("time.sleep", return_value=None)
     def test_moderator_invite__no_perms(self, _):
@@ -2131,7 +2131,7 @@ class TestSubredditStylesheet(IntegrationTest):
                 self.subreddit.stylesheet.upload(
                     "praw", self.image_path("invalid.jpg")
                 )
-        assert excinfo.value.error_type == "IMAGE_ERROR"
+        assert excinfo.value[0].error_type == "IMAGE_ERROR"
 
     def test_upload__invalid_ext(self):
         self.reddit.read_only = False
@@ -2142,7 +2142,7 @@ class TestSubredditStylesheet(IntegrationTest):
                 self.subreddit.stylesheet.upload(
                     "praw.png", self.image_path("white-square.png")
                 )
-        assert excinfo.value.error_type == "BAD_CSS_NAME"
+        assert excinfo.value[0].error_type == "BAD_CSS_NAME"
 
     def test_upload__too_large(self):
         self.reddit.read_only = False
@@ -2287,7 +2287,7 @@ class TestSubredditStylesheet(IntegrationTest):
                     getattr(self.subreddit.stylesheet, method)(
                         self.image_path("invalid.jpg")
                     )
-                assert excinfo.value.error_type == "IMAGE_ERROR"
+                assert excinfo.value[0].error_type == "IMAGE_ERROR"
 
     def test_upload__others_too_large(self):
         self.reddit.read_only = False

--- a/tests/integration/test_objector.py
+++ b/tests/integration/test_objector.py
@@ -15,5 +15,5 @@ class TestObjector(IntegrationTest):
             submission = self.reddit.submission("4b536h")
             with pytest.raises(APIException) as excinfo:
                 submission.mod.approve()
-            assert excinfo.value.error_type == "USER_REQUIRED"
-            assert str(excinfo.value) == message
+            assert excinfo.value[0].error_type == "USER_REQUIRED"
+            assert str(excinfo.value[0]) == message

--- a/tests/integration/test_objector.py
+++ b/tests/integration/test_objector.py
@@ -1,7 +1,6 @@
 """Test praw.objector."""
+from praw.exceptions import RedditAPIException
 import pytest
-
-from praw.exceptions import APIException
 
 from . import IntegrationTest
 
@@ -13,7 +12,7 @@ class TestObjector(IntegrationTest):
             "TestObjector.test_raise_api_exception"
         ):
             submission = self.reddit.submission("4b536h")
-            with pytest.raises(APIException) as excinfo:
+            with pytest.raises(RedditAPIException) as excinfo:
                 submission.mod.approve()
-            assert excinfo.value[0].error_type == "USER_REQUIRED"
-            assert str(excinfo.value[0]) == message
+            assert excinfo.value.items[0].error_type == "USER_REQUIRED"
+            assert str(excinfo.value.items[0]) == message

--- a/tests/integration/test_objector.py
+++ b/tests/integration/test_objector.py
@@ -1,6 +1,7 @@
 """Test praw.objector."""
-from praw.exceptions import RedditAPIException
 import pytest
+
+from praw.exceptions import RedditAPIException
 
 from . import IntegrationTest
 

--- a/tests/unit/test_deprecations.py
+++ b/tests/unit/test_deprecations.py
@@ -17,6 +17,7 @@ class TestDeprecation(UnitTest):
         self.reddit.validate_on_submit = False
         with pytest.raises(DeprecationWarning):
             self.reddit.validate_on_submit
+
     def test_apiexception(self):
         exc = APIException(["test", "testing", "test"])
         with pytest.raises(DeprecationWarning):

--- a/tests/unit/test_deprecations.py
+++ b/tests/unit/test_deprecations.py
@@ -2,6 +2,8 @@
 
 import pytest
 
+from praw.exceptions import APIException
+
 from . import UnitTest
 
 
@@ -15,3 +17,11 @@ class TestDeprecation(UnitTest):
         self.reddit.validate_on_submit = False
         with pytest.raises(DeprecationWarning):
             self.reddit.validate_on_submit
+    def test_apiexception(self):
+        exc = APIException(["test", "testing", "test"])
+        with pytest.raises(DeprecationWarning):
+            exc.error_type
+        with pytest.raises(DeprecationWarning):
+            exc.message
+        with pytest.raises(DeprecationWarning):
+            exc.field

--- a/tests/unit/test_exceptions.py
+++ b/tests/unit/test_exceptions.py
@@ -1,6 +1,8 @@
 # coding: utf-8
+import pytest
+
 from praw.exceptions import (
-    RedditAPIException,
+    APIException,
     ClientException,
     DuplicateReplaceException,
     InvalidFlairTemplateID,
@@ -8,6 +10,7 @@ from praw.exceptions import (
     InvalidURL,
     MissingRequiredAttributeException,
     PRAWException,
+    RedditAPIException,
     RedditErrorItem,
     WebSocketException,
 )
@@ -74,6 +77,15 @@ class TestRedditAPIException:
         )
         for exception in container.items:
             assert isinstance(exception, RedditErrorItem)
+
+    @pytest.mark.filterwarnings("ignore", category=DeprecationWarning)
+    def test_apiexception_value(self):
+        exc = APIException("test", "testing", "test")
+        assert exc.error_type == "test"
+        exc2 = APIException(["test", "testing", "test"])
+        assert exc2.message == "testing"
+        exc3 = APIException([["test", "testing", "test"]])
+        assert exc3.field == "test"
 
 
 class TestClientException:

--- a/tests/unit/test_exceptions.py
+++ b/tests/unit/test_exceptions.py
@@ -1,6 +1,6 @@
 # coding: utf-8
 from praw.exceptions import (
-    APIException,
+    RedditAPIException,
     ClientException,
     DuplicateReplaceException,
     InvalidFlairTemplateID,
@@ -28,7 +28,6 @@ class TestRedditErrorItem:
         error = RedditErrorItem(*resp)
         error2 = RedditErrorItem(*resp)
         assert error == error2
-        assert error == resp
         assert error != 0
 
     def test_property(self):
@@ -55,16 +54,17 @@ class TestRedditErrorItem:
         )
         assert (
             repr(error)
-            == "BAD_SOMETHING: 'invalid something' on field 'some_field'"
+            == "RedditErrorItem(error_type='BAD_SOMETHING', message="
+            "'invalid something', field='some_field')"
         )
 
 
-class TestAPIException:
+class TestRedditAPIException:
     def test_inheritance(self):
-        assert isinstance(APIException([[None, None, None]]), PRAWException)
+        assert issubclass(RedditAPIException, PRAWException)
 
-    def test_iter(self):
-        container = APIException(
+    def test_items(self):
+        container = RedditAPIException(
             [
                 ["BAD_SOMETHING", "invalid something", "some_field"],
                 RedditErrorItem(
@@ -72,19 +72,8 @@ class TestAPIException:
                 ),
             ]
         )
-        for exception in container:
+        for exception in container.items:
             assert isinstance(exception, RedditErrorItem)
-
-    def test_getitem(self):
-        container = APIException(
-            [
-                ["BAD_SOMETHING", "invalid something", "some_field"],
-                ["BAD_SOMETHING", "invalid something", "some_field"],
-            ]
-        )
-        assert isinstance(container[0], RedditErrorItem)
-        assert isinstance(container[-1], RedditErrorItem)
-        assert isinstance(container[0:2], list)
 
 
 class TestClientException:

--- a/tests/unit/test_exceptions.py
+++ b/tests/unit/test_exceptions.py
@@ -62,6 +62,13 @@ class TestRedditErrorItem:
         )
 
 
+class TestAPIException:
+    def test_catch(self):
+        exc = RedditAPIException([["test", "testing", "test"]])
+        with pytest.raises(APIException):
+            raise exc
+
+
 class TestRedditAPIException:
     def test_inheritance(self):
         assert issubclass(RedditAPIException, PRAWException)
@@ -80,11 +87,11 @@ class TestRedditAPIException:
 
     @pytest.mark.filterwarnings("ignore", category=DeprecationWarning)
     def test_apiexception_value(self):
-        exc = APIException("test", "testing", "test")
+        exc = RedditAPIException("test", "testing", "test")
         assert exc.error_type == "test"
-        exc2 = APIException(["test", "testing", "test"])
+        exc2 = RedditAPIException(["test", "testing", "test"])
         assert exc2.message == "testing"
-        exc3 = APIException([["test", "testing", "test"]])
+        exc3 = RedditAPIException([["test", "testing", "test"]])
         assert exc3.field == "test"
 
 

--- a/tests/unit/test_exceptions.py
+++ b/tests/unit/test_exceptions.py
@@ -8,6 +8,7 @@ from praw.exceptions import (
     InvalidURL,
     MissingRequiredAttributeException,
     PRAWException,
+    RedditErrorItem,
     WebSocketException,
 )
 
@@ -21,17 +22,69 @@ class TestPRAWException:
         assert str(PRAWException("foo")) == "foo"
 
 
-class TestAPIException:
-    def test_inheritance(self):
-        assert isinstance(APIException(None, None, None), PRAWException)
+class TestRedditErrorItem:
+    def test_equality(self):
+        resp = ["BAD_SOMETHING", "invalid something", "some_field"]
+        error = RedditErrorItem(*resp)
+        error2 = RedditErrorItem(*resp)
+        assert error == error2
+        assert error == resp
+        assert error != 0
+
+    def test_property(self):
+        error = RedditErrorItem(
+            *["BAD_SOMETHING", "invalid something", "some_field"]
+        )
+        assert (
+            error.error_message
+            == "BAD_SOMETHING: 'invalid something' on field 'some_field'"
+        )
 
     def test_str(self):
-        exception = APIException(
-            "BAD_SOMETHING", "invalid something", "some_field"
+        error = RedditErrorItem(
+            *["BAD_SOMETHING", "invalid something", "some_field"]
         )
-        assert str(exception) == (
-            "BAD_SOMETHING: 'invalid something' on field 'some_field'"
+        assert (
+            str(error)
+            == "BAD_SOMETHING: 'invalid something' on field 'some_field'"
         )
+
+    def test_repr(self):
+        error = RedditErrorItem(
+            *["BAD_SOMETHING", "invalid something", "some_field"]
+        )
+        assert (
+            repr(error)
+            == "BAD_SOMETHING: 'invalid something' on field 'some_field'"
+        )
+
+
+class TestAPIException:
+    def test_inheritance(self):
+        assert isinstance(APIException([[None, None, None]]), PRAWException)
+
+    def test_iter(self):
+        container = APIException(
+            [
+                ["BAD_SOMETHING", "invalid something", "some_field"],
+                RedditErrorItem(
+                    *["BAD_SOMETHING", "invalid something", "some_field"]
+                ),
+            ]
+        )
+        for exception in container:
+            assert isinstance(exception, RedditErrorItem)
+
+    def test_getitem(self):
+        container = APIException(
+            [
+                ["BAD_SOMETHING", "invalid something", "some_field"],
+                ["BAD_SOMETHING", "invalid something", "some_field"],
+            ]
+        )
+        assert isinstance(container[0], RedditErrorItem)
+        assert isinstance(container[-1], RedditErrorItem)
+        assert isinstance(container[0:2], list)
 
 
 class TestClientException:

--- a/tests/unit/test_exceptions.py
+++ b/tests/unit/test_exceptions.py
@@ -35,7 +35,7 @@ class TestRedditErrorItem:
 
     def test_property(self):
         error = RedditErrorItem(
-            *["BAD_SOMETHING", "invalid something", "some_field"]
+            "BAD_SOMETHING", "invalid something", "some_field"
         )
         assert (
             error.error_message
@@ -44,7 +44,7 @@ class TestRedditErrorItem:
 
     def test_str(self):
         error = RedditErrorItem(
-            *["BAD_SOMETHING", "invalid something", "some_field"]
+            "BAD_SOMETHING", "invalid something", "some_field"
         )
         assert (
             str(error)
@@ -53,7 +53,7 @@ class TestRedditErrorItem:
 
     def test_repr(self):
         error = RedditErrorItem(
-            *["BAD_SOMETHING", "invalid something", "some_field"]
+            "BAD_SOMETHING", "invalid something", "some_field"
         )
         assert (
             repr(error)
@@ -78,7 +78,7 @@ class TestRedditAPIException:
             [
                 ["BAD_SOMETHING", "invalid something", "some_field"],
                 RedditErrorItem(
-                    *["BAD_SOMETHING", "invalid something", "some_field"]
+                    "BAD_SOMETHING", "invalid something", "some_field"
                 ),
             ]
         )

--- a/tests/unit/test_objector.py
+++ b/tests/unit/test_objector.py
@@ -1,6 +1,6 @@
 import pytest
 
-from praw.exceptions import APIException, ClientException
+from praw.exceptions import RedditAPIException, ClientException
 
 from . import UnitTest
 
@@ -23,7 +23,7 @@ class TestObjector(UnitTest):
             }
         }
         error = objector.parse_error(error_response)
-        assert isinstance(error, APIException)
+        assert isinstance(error, RedditAPIException)
 
         error_response = {"json": {"errors": []}}
         with pytest.raises(ClientException):
@@ -51,5 +51,5 @@ class TestObjector(UnitTest):
                 ]
             }
         }
-        with pytest.raises(APIException):
+        with pytest.raises(RedditAPIException):
             objector.check_error(error_response)

--- a/tests/unit/test_objector.py
+++ b/tests/unit/test_objector.py
@@ -1,6 +1,6 @@
 import pytest
 
-from praw.exceptions import RedditAPIException, ClientException
+from praw.exceptions import ClientException, RedditAPIException
 
 from . import UnitTest
 
@@ -37,8 +37,9 @@ class TestObjector(UnitTest):
                 ]
             }
         }
-        with pytest.raises(AssertionError):
-            objector.parse_error(error_response)
+        assert isinstance(
+            objector.parse_error(error_response), RedditAPIException
+        )
 
     def test_check_error(self):
         objector = self.reddit._objector

